src/bundler.ts

```diff
@@ -81,52 +110,155 @@ export class Bundler {
                 await fs.access(fullPath);
                 importData.found = true;
             } catch (error) {
-                let underscoredDirname = path.dirname(fullPath);
-                let underscoredBasename = path.basename(fullPath);
-                let underscoredFilePath = path.join(underscoredDirname, `_${underscoredBasename}`);
+                const underscoredDirname = path.dirname(fullPath);
+                const underscoredBasename = path.basename(fullPath);
+                const underscoredFilePath = path.join(underscoredDirname, `_${underscoredBasename}`);
                 try {
                     await fs.access(underscoredFilePath);
                     importData.fullPath = underscoredFilePath;
                     importData.found = true;
                 } catch (underscoreErr) {
                     // Neither file, nor partial was found
+                    // Skipping...
                 }
             }
 
             return importData;
         });
 
-        let imports = await Promise.all(importsPromises);
-        let allImports: BundleResult[] = [];
-        for (let imp of imports) {
+        // Wait for all imports file names to be resolved
+        const imports = await Promise.all(importsPromises);
+
+        const bundleResult: BundleResult = {
+            rootBundleResult: rootBundleResult,
+            filePath: filePath,
+            found: true
+        };
+
+        // if this is the root BundleResult
+        if (rootBundleResult == null) {
+            // Initialize usedImports array
+            bundleResult.usedImports = {};
+            bundleResult.importsByFile = {};
+        }

```

Comment:
Instead of saving into rootBundleResult, you can remove static methods and use class properties instead?